### PR TITLE
summer internship 2024

### PR DESCRIPTION
**Pull request by trel**

### CODE REVIEW

**Comment by korydraughn** [comment]
On a first pass, everything reads well to me. Will read it again tomorrow and see if that thought changes.

File: content/posts/irods-internship-summer-2024.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+[https://github.com/irods](https://github.com/irods)
+
+## Five Projects
+
+In no particular order:
+
+
+### 1) Create an iRODS database plugin template repository (C++)
+
+The iRODS Development Team has built project templates as examples for four of the seven plugin interfaces into iRODS and one more as an example C++ client.  The next plugin we'd like to have a template for is the database plugin.  iRODS currently supports PostgreSQL, MySQL/MariaDB, and Oracle, but the code has not been pulled out into separate repositories, yet.  This example repository would pave the way for increased modularity of the codebase.
```
Comment (korydraughn) [inline content/posts/irods-internship-summer-2024.md:34]: An even bigger benefit would be the ability for devs (us included) to experiment adding support for other databases and systems to iRODS. I feel that is the primary focus of the template project effort. _I'm looking at the last statement specifically here._

I don't know that we'd ever separate the three backends we currently support, but I can see us rewriting the implementation using nanodbc and a couple if-stmts to handle database specific behavior.

**Review by korydraughn — COMMENTED** [review]
Seems good to me.

Just one thing about the database plugin template project.

File: content/posts/irods-internship-summer-2024.md (hunk 1 above)
Comment (trel) [reply content/posts/irods-internship-summer-2024.md:34]: ah, very good.  will reword a bit.

Comment (trel) [reply content/posts/irods-internship-summer-2024.md]: reworded in second commit.

Comment (korydraughn) [reply content/posts/irods-internship-summer-2024.md]: Looks good.

Comment (trel) [reply content/posts/irods-internship-summer-2024.md]: good... squashing.

**Pull request merged**